src/main/java/org/example/DoWhileLoopMatrixGenerator.java

```diff
@@ -0,0 +1,39 @@
+package org.example;
+
+
+import static java.lang.Math.abs;
+
+public class DoWhileLoopMatrixGenerator {
+    public static void generateAndPrintMatrixWithDoWhileLoop(int n) {
+        int[][] matrix = generateMatrixWithDoWhileLoop(n);
+        printMatrix(matrix);
+    }
+    private static int[][] generateMatrixWithDoWhileLoop(int n) {
+        int i = 0;
+        int j = 0;
+        int[][] matrix = new int[n][n];
+        // generating and filling the matrix
+        do {
+            do {
+                matrix[i][j] = abs(i - j);
+                j++;
+            } while (j < n);
+            i++;
+            j = 0;
+        } while (i < n);
+        return matrix;
+    }
+    private static void printMatrix(int[][] matrix) {
+        int i = 0;
+        int j = 0;
+        do {
+            do {
+                System.out.print(matrix[i][j] + " ");
+                j++;
+            } while (j < matrix.length);
```

Comment:
notice your both while conditions which rely on `matrix.lenght`. This won't work if your matrix is not square.
I would agree that `generateMatrixWithDoWhileLoop` knows that the matrix is square because it creates it. But your `printMatrix` looks like a generic print which has to print all dimensional matrixes.
So you have some options. Or you fix the logic so that it prints any matrix. Or you rename your print method to `printSquareMatrix` to let users know that it prints only square matrixes. If you choose the second option with rename, then before proceeeding to printing logic you have to check if the matrix provided as parameter is really square. If not, throw exception like `IllegalArgumentException`

Reply:
I have added a solution for this, but the scope actually is only generating and printing square matrixes

Reply:
😄  I understand. That's why I've added an option for square matrixes only. 
I am trying to show you how would a reviewer react to your changes in a real project. And also showing that you have to protect your code for example from unexpected user input and help user to understand his mistakes in case he inputs incorrect parameters